mobile/native/src/dlc/node.rs

```diff
@@ -517,21 +531,68 @@ impl Node {
-    pub fn process_settle_offer(&self, channel_id: &DlcChannelId) -> Result<()> {
+    #[instrument(fields(channel_id = hex::encode(offer.settle_offer.channel_id)),skip_all, err(Debug))]
+    pub fn process_settle_offer(&self, offer: &TenTenOneSettleOffer) -> Result<()> {
         // TODO(holzeis): We should check if the offered amounts are expected.
+        let order_reason = offer.order.clone().order_reason;
+        let order_id = offer.order.id;
+
+        match order_reason {
+            OrderReason::Expired
+            | OrderReason::CoordinatorLiquidated
+            | OrderReason::TraderLiquidated => {
+                tracing::info!(
+                    %order_id,
+                    "Received an async match from orderbook. Reason: {order_reason:?}"
+                );
 
+                event::publish(&EventInternal::BackgroundNotification(
+                    BackgroundTask::AsyncTrade(order_reason.into()),
+                ));
+
+                order::handler::async_order_filling(&offer.order, &offer.filled_with)
+                    .with_context(|| {
+                        format!(
+                        "Failed to process async match update from orderbook. order_id {order_id}"
+                    )
+                    })?;
+            }
+            // Received a regular settle offer after a manual order.
+            //
+            // TODO(holzeis): Eventually this should as well start the trade dialog. At the moment
+            // we automatically show the trade dialog since we expect a synchronous response from
+            // the orderbook.
+            OrderReason::Manual => self.set_order_to_filling(offer.filled_with.clone())?,
+        }
+
+        let channel_id = offer.settle_offer.channel_id;
         if let Err(e) = self
             .inner
-            .accept_dlc_channel_collaborative_settlement(channel_id)
+            .accept_dlc_channel_collaborative_settlement(&channel_id)
         {
             tracing::error!("Failed to accept dlc channel collaborative settlement offer. {e:#}");
-            self.reject_settle_offer(channel_id)?;
+            self.reject_settle_offer(&channel_id)?;
         }
 
         Ok(())
     }
 
+    fn set_order_to_filling(&self, filled_with: commons::FilledWith) -> Result<()> {
+        let order_id = filled_with.order_id;
+        tracing::info!(%order_id, "Received match from orderbook");
+        let execution_price = filled_with
+            .average_execution_price()
+            .to_f32()
+            .expect("to fit into f32");
+
+        let matching_fee = filled_with.order_matching_fee();
+
+        // TODO(holzeis): Add handling to be more resilient if the order could not be found.
+        order::handler::order_filling(order_id, execution_price, matching_fee).with_context(|| {
+            format!("Failed to process match update from orderbook. order_id = {order_id}")
+        })
```

Comment:
🔧 I disagree with the TODO.

If the app doesn't know about the order match I don't think it should trust the coordinator. At the moment we don't even verify that the created order and the matched order are the same beyond the ID, which is already problematic.

Reply:
The todo is probably not clear, but this is actually for the scenario where the order exists but not in the expected state.

Reply:
Ah, fair enough. That makes sense then.

Reply:
Actually you've been right.. we use the order id here already so that should be fine. Nothing to do.

Reply:
Ah, we don't actually check the original state when we set to `Filling`! Nice then.